File: src/api/java/com/ultreon/mods/advanceddebug/api/common/IFormattable.java
package com.ultreon.mods.advanceddebug.api.common;

import com.ultreon.mods.advanceddebug.api.client.formatter.IFormatterContext;

public interface IFormattable {
    @SuppressWarnings("DeprecatedIsStillUsed")
    @Deprecated(forRemoval = true)
    default String toFormattedString() {
        throw new AssertionError("toFormattedString(IFormatterContext) is not implemented");
    }

    @SuppressWarnings("deprecation")
    default void format(IFormatterContext ctx) {
        ctx.direct(toFormattedString());
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/api/common/IFormattable.kt
package com.ultreon.mods.advanceddebug.api.common

import com.ultreon.mods.advanceddebug.api.client.formatter.IFormatterContext

interface IFormattable {
    @Deprecated("")
    fun toFormattedString(): String {
        throw AssertionError("toFormattedString(IFormatterContext) is not implemented")
    }

    @Suppress("DEPRECATION")
    fun format(ctx: IFormatterContext) {
        ctx.direct(toFormattedString())
    }
}